lib/handler/debug_state.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2016 DeNA Co., Ltd., Ichito Nagata
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to
+ * deal in the Software without restriction, including without limitation the
+ * rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+ * sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+ * IN THE SOFTWARE.
+ */
+#include "h2o.h"
+
+#include <inttypes.h>
+#include "h2o/http2.h"
+
+struct st_h2o_debug_state_handler_t {
+    h2o_handler_t super;
+    int hpack_enabled : 1;
+};
+
+static int on_req(h2o_handler_t *_self, h2o_req_t *req)
+{
+    struct st_h2o_debug_state_handler_t *self = (void *)_self;
+
+    static h2o_generator_t generator = {NULL, NULL};
+
+    h2o_http2_debug_state_t *debug_state = h2o_get_http2_debug_state(req, self->hpack_enabled);
+    if (debug_state == NULL) {
+        req->res.status = 404;
+        h2o_add_header(&req->pool, &req->res.headers, H2O_TOKEN_CONTENT_TYPE, H2O_STRLIT("text/plain; charset=utf-8"));
+        h2o_add_header(&req->pool, &req->res.headers, H2O_TOKEN_CACHE_CONTROL, H2O_STRLIT("no-cache, no-store"));
+        h2o_start_response(req, &generator);
+        h2o_send(req, NULL, 0, 1);
+        return 0;
+    }
+
+    const h2o_token_t conn_flow_in_token = { { H2O_STRLIT("conn-flow-in") } };
+    const h2o_token_t conn_flow_out_token = { { H2O_STRLIT("conn-flow-out") } };
```

Comment:
You cannot do this.

`h2o_token_t` is an object that be compared by checking their address. To achieve such property, a string contained by a token MUST be interned (i.e. only one `h2o_token_t` should exist within the entire program for each string). Trying to define a token in each source file collides with the requirement.

You should instead use `h2o_add_header_by_str` for registering a header that is (either guaranteed or might) not be a token.